### ISSUE TITLE
prefer-object-from-entries

### ISSUE THREAD

**Issue by GuoSirius** [sindresorhus/eslint-plugin-unicorn#1464]
Prefer `Object.fromEntries()` over `Array#reduce()`  unicorn/prefer-object-from-entries

```js
const demo= [{a: 1 b:3, c: 3},{a: 11 b:13, c: 13}]

const reduced = demo.reduce((acc, item) => {
  acc[item.a] = item.b

  return acc
}, {})
```

**Comment by fisker** [comment]
What is the problem?

**Comment by alfaproject** [comment]
Those are not key/value pairs. They are just a list of objects. I've also have a lot of false positives

This is a grouping reduce by a property of those objects (groupBy)

**Comment by fisker** [comment]
```js
const reduced = Object.fromEntries(
  demo.map(item => [
    item.a, // key
    item.b, // value
  ),
);
```

**Comment by alfaproject** [comment]
So, you are going to really suggest creating a new array just to have key/value pairs instead of making this rule optional by default or add some option. Wow efficient...

**Comment by fisker** [comment]
You can disable this rule `{'unicorn/prefer-object-from-entries': 'off'}`, if you don't like it.

**Comment by alfaproject** [comment]
Already did but that's not the point. If it's not easy to detect arrays that are pairs I get it, that's fine. It sounds weird to me tho that you would suggest mapping but sure, I guess. I noticed there's another issue to track improvements so I'll subscribe to that in the meanwhile

**Comment by fisker** [comment]
I understand that there might be sight performance loss, but the second one is more readable to me.

<details>
<summary>I think this is acceptable.</summary>

```js
const array = Array.from({length: 1e2}, () => ['k', 'v'])

console.time('reduce');
for (let i = 0; i < 1e3; i++)
array.reduce((acc, item) => {
  acc[item[0]] = item[1]

  return acc
}, {})
console.timeEnd('reduce');

console.time('fromEntries');
for (let i = 0; i < 1e3; i++)
Object.fromEntries(
  array.map(item => [
    item[0], // key
    item[1], // value
  ]),
);
console.timeEnd('fromEntries');
VM141:10 reduce: 4.244873046875 ms
VM141:20 fromEntries: 6.815185546875 ms
```

</details>


I don't think there are not too many cases, the array is already key-value pairs, most of them need transform.  Any real word cases can help us to understand why you don't like it? 

The one in the topic seems silly to me, if there are duplicated value of `a` property, it may result unexpected.

**Comment by fregante** [comment]
Better yet, no functions at all:

```js
const demo= [{a: 1 b:3, c: 3},{a: 11 b:13, c: 13}]

const reduced = {};
for (const item of demo) {
	reduced[item.a] = item.b
}
```

<details>
<summary>Perf test</summary>

```js
const array = Array.from({length: 1e2}, () => ['k', 'v'])

console.time('reduce');
for (let i = 0; i < 1e3; i++)
array.reduce((acc, item) => {
  acc[item[0]] = item[1]

  return acc
}, {})
console.timeEnd('reduce');

console.time('fromEntries');
for (let i = 0; i < 1e3; i++)
Object.fromEntries(
  array.map(item => [
    item[0], // key
    item[1], // value
  ]),
);
console.timeEnd('fromEntries');


console.time('for-of');
for (let i = 0; i < 1e3; i++) {
    const reduced = {};
    for (const item of array) {
        reduced[item[0]] = item[1]
    }
}
console.timeEnd('for-of');
VM237:10 reduce: 3.24609375 ms
VM237:20 fromEntries: 13.962158203125 ms
VM237:30 for-of: 5.65380859375 ms
```

</details>
